src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Aot.targets

```diff
@@ -48,8 +48,27 @@ They run in a context of an inner build with a single $(RuntimeIdentifier).
       <FileWrites Include="@(_AotCompiledAssemblies)" />
       <ResolvedFileToPublish
           Include="@(_AotCompiledAssemblies)"
+          Condition=" '$(_AndroidUseMarshalMethods)' != 'True' "
           ArchiveFileName="libaot-$([System.IO.Path]::GetFileNameWithoutExtension('%(_AotCompiledAssemblies.Identity)')).so"
       />
+      <_AotResolvedFileToPublish
+          Include="@(_AotCompiledAssemblies)"
+          Condition=" '$(_AndroidUseMarshalMethods)' == 'True' "
+          ArchiveFileName="libaot-$([System.IO.Path]::GetFileNameWithoutExtension('%(_AotCompiledAssemblies.Identity)')).so"
+      />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_AndroidAotAndComputeFilesToPublishForRuntimeIdentifiers"
+          DependsOnTargets="_AndroidAot"
+          Returns="@(_AotResolvedFileToPublish)">
+    <ItemGroup>
+      <_AotResolvedFileToPublish Remove="@(_SourceItemsToCopyToPublishDirectory)" />
+      <_AotResolvedFileToPublish
+          Condition=" '%(_AotResolvedFileToPublish.RuntimeIdentifier)' == '' "
+          Update="@(_AotResolvedFileToPublish)"
+          RuntimeIdentifier="$(RuntimeIdentifier)"
+      />
```

Comment:
It is converted to `ResolvedFileToPublish` here https://github.com/xamarin/xamarin-android/pull/7351/files#diff-e2c60a31385c6fbadf9b9f439d3681ee357077bf3e66f181a0ddb02b9d98adf1R1668, while if it returned `@(ResolvedFileToPublish)`, it would duplicate the items that were in the group **before** the target was invoked.

Reply:
```xml
<Target Name="_RunAotForAllRIDs"
        Condition=" '$(_AndroidUseMarshalMethods)' == 'True' "
        Returns="@(ResolvedFileToPublish)">
  <MSBuild
      Projects="@(_ProjectToBuild)"
      BuildInParallel="$(_AndroidBuildRuntimeIdentifiersInParallel)"
      Targets="_AndroidAotAndComputeFilesToPublishForRuntimeIdentifiers">
    <Output TaskParameter="TargetOutputs" ItemName="ResolvedFileToPublish" />
  </MSBuild>
</Target>
```

The `<Output TaskParameter="TargetOutputs">` expression doesn't care what is the name of items returned by the called target - it just takes the items and copies them with the new name specified in `ItemName`. And since we use `_AotResolvedFileToPublish` in the `Returns` attribute of the `_AndroidAotAndComputeFilesToPublishForRuntimeIdentifiers` target, we don't get duplicates of `ResolvedFileToPublish`

Reply:
Ok, I see it now. 👍